contracts/HoprStake2.sol

```diff
@@ -0,0 +1,443 @@
+
+    address public LOCK_TOKEN = 0xD057604A14982FE8D88c5fC25Aac3267eA142a08; // Token that HOPR holders need to lock to the contract: xHOPR address.
+    address public REWARD_TOKEN = 0xD4fdec44DB9D44B8f2b6d529620f9C0C7066A2c1; // Token that HOPR holders can claim as rewards: wxHOPR address
+    IHoprBoost public NFT_CONTRACT = IHoprBoost(0x43d13D7B83607F14335cF2cB75E87dA369D056c7) ; // Address of the HoprBoost NFT smart contract.
+
+    mapping(address=>mapping(uint256=>uint256)) public redeemedNft; // Redeemed NFT per account, structured as “account -> index -> NFT tokenId”.
+    mapping(address=>uint256) public redeemedNftIndex; // The last index of redeemed NFT of an account. It defines the length of the “redeemedBoostToken mapping.
+    mapping(address=>mapping(uint256=>uint256)) public redeemedFactor; // Redeemed boost factor per account, structured as “account -> index -> NFT tokenId”.
+    mapping(address=>uint256) public redeemedFactorIndex; // The last index of redeemed boost factor factor of an account. It defines the length of the “redeemedFactor” mapping.
+
+    mapping(address=>Account) public accounts; // It stores the locked token amount, earned and claimed rewards per account.
+    uint256 public totalLocked;  // Total amount of tokens being locked in the incentive program.
+    uint256 public availableReward; // Total amount of reward tokens currently available in the lock.
+
+    // setup ERC1820
+    IERC1820Registry private constant ERC1820_REGISTRY = IERC1820Registry(0x1820a4B7618BdE71Dce8cdc73aAB6C95905faD24);
+    bytes32 private constant TOKENS_RECIPIENT_INTERFACE_HASH = keccak256("ERC777TokensRecipient");
+
+    event Sync(address indexed account, uint256 indexed increment);
+    event Staked(address indexed account, uint256 indexed actualAmount);
+    event Released(address indexed account, uint256 indexed actualAmount);
+    event RewardFueled(uint256 indexed amount);
+    event Redeemed(address indexed account, uint256 indexed boostTokenId, bool indexed factorRegistered);
+    event Claimed(address indexed account, uint256 indexed rewardAmount);
+
+    /**
+     * @dev Provide NFT contract address. Transfer owner role to the new owner address. 
+     * At deployment, it also registers the lock contract as an ERC777 recipient.
+     * @param _nftAddress address Address of the NFT contract.
+     * @param _newOwner address Address of the new owner. This new owner can reclaim any ERC20 and ERC721 token being accidentally sent to the lock contract. 
+     * @param _lockToken address Address of the stake token xHOPR.
+     * @param _rewardToken address Address of the reward token wxHOPR.
+     */
+    constructor(address _nftAddress, address _newOwner, address _lockToken, address _rewardToken) {
+        // implement in favor of testing
+        uint chainId;
+        assembly {
+            chainId := chainid()
+        }
+        if (chainId != 100) {
```

Comment:
actually why `!=`? isnt xDAI 100...?

Reply:
oh I see you got default vars. So I guess these constructor parameters are just for dev/debugging, right?

Reply:
Yes. Just for tests to run through